src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -74,12 +74,14 @@ def _learn(self) -> List[PartialNSRTAndDatastore]:
         # to learn PNADs. Repeat until a fixed point is reached.
         nec_pnad_set_changed = True
         while nec_pnad_set_changed:
-            # Before each pass, clear the poss_keep_effects of all the
-            # PNADs. We do this because we only want the poss_keep_effects
-            # of the final pass, where the PNADs did not change.
+            # Before each pass, clear the poss_keep_effects and
+            # seg_to_keep_effects_sub of all the PNADs. We do this because
+            # we only want the poss_keep_effects of the final pass, where
+            # the PNADs did not change.
             for pnads in param_opt_to_nec_pnads.values():
                 for pnad in pnads:
                     pnad.poss_keep_effects = set()
+                    pnad.seg_to_keep_effects_sub = {}
```

Comment:
NIT: could we do `pnad.seg_to_keep_effects_sub.clear()`?

Reply:
yeah good suggestion